### PR TITLE
refactor(pinia): migrate breakoutRoomsStore to Pinia 🍍

### PR DESCRIPTION
**Pull request by Antreesy**
* Migrate breakoutRoomsStore to Pinia :pineapple: 
  * optimize some methods
  * rename **Action to normal ones

### 🖼️ Screenshots / Screencasts

No visual changes

### 🚧 Tasks

- [x] Pinia migration
- [ ] Typescript migration - in follow-up
- [ ] Test coverage - in follow-up

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] ⛑️ Tests are included or not possible

### CODE REVIEW

File: src/stores/breakoutRooms.js [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+		 * @param {Array<object>|object} conversationOrArray a single conversation or an array of conversations.
+		 *
+		 */
+		processConversations(token, conversationOrArray) {
+			const conversations = Array.isArray(conversationOrArray) ? conversationOrArray : [conversationOrArray]
+			conversations.forEach(conversation => {
+				if (conversation.objectType === CONVERSATION.OBJECT_TYPE.BREAKOUT_ROOM) {
+					this.addBreakoutRoom(token, conversation)
+				}
+			})
```
Comment (DorraJaouad) [inline src/stores/breakoutRooms.js:80]: Already in `'patchConversations'`

File: src/stores/breakoutRooms.js [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+			try {
+				await broadcastMessageToBreakoutRooms(token, message)
+			} catch (error) {
+				console.error(error)
+				showError(t('spreed', 'An error occurred while sending a message to the breakout rooms'))
+			}
+		},
+
+		/**
+		 * Update a participants in breakout rooms for a given token.
```
Comment (DorraJaouad) [inline src/stores/breakoutRooms.js:235]:
```suggestion
		 * Update participants in breakout rooms for a given token.
```

File: src/stores/breakoutRooms.js [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+		 *
+		 * @param {string} token the parent room token.
+		 */
+		async fetchBreakoutRoomsParticipants(token) {
+			try {
+				const response = await getBreakoutRoomsParticipants(token)
```
Comment (DorraJaouad) [inline src/stores/breakoutRooms.js:241]:
```suggestion
				const response = await fetchBreakoutRoomsParticipants(token)
```

Rename this to `fetchBreakoutRoomsParticipants` ?

File: src/stores/breakoutRooms.js [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+		},
+
+		/**
+		 * Dismiss a notification about raised hand for a breakout room with given token.
+		 *
+		 * @param {string} token the breakout room token.
+		 */
+		async dismissRequestAssistance(token) {
+			try {
+				const response = await resetRequestAssistance(token)
```
Comment (DorraJaouad) [inline src/stores/breakoutRooms.js:281]: same

**Review by DorraJaouad — COMMENTED** [review]
Testing : 
image
Participants are not in their right room.

Also, is it expected to return room-not-found when reassigning a participant to the room where you are currently joining the call?

File: src/stores/breakoutRooms.js (hunk 1 above)
Comment (Antreesy) [reply src/stores/breakoutRooms.js:80]: Removed

**Comment by Antreesy** [comment]
> Participants are not in their right room

Good catch, they might be received from different sources

> is it expected to return room-not-found when reassigning a participant to the room

Should be expected, as participant doesn't have access rights to the previous room anymore

**Comment by DorraJaouad** [comment]
> Should be expected, as participant doesn't have access rights to the previous room anymore

* Alice is in call in room 1
* John in room 2 and moderator moves him to room 1 
* Alice gets page-not-found

**Comment by Antreesy** [comment]
> * John in room 2 and moderator moves him to room 1
> * Alice gets page-not-found

I'd sat that's on API side. If you check system messages, you'll see that everyone got removed from BR, then only selected participants are added back

**Review by DorraJaouad — APPROVED** [review]
As follow-ups are there, this is ready

**Pull request merged**